portfolio/src/main/webapp/calendar.js

```diff
@@ -151,15 +152,173 @@ function collectAllEvents() {
+ * The parameters are of type String. 
+ */
+function parseTodayString(year, month, date) {
+  // Adds leading 0 as padding for month and date strings.
+  if (parseInt(month) < 10) {
+    month = '0' + month;
+  }
+  
+  if (parseInt(date) < 10) {
+    date = '0' + date;
+  }
+  const todayString = year + '-' + month + '-' + date;
+  return todayString; 
+}
+ 
+/**
+ * Sets the default calendar event start and times.
+ * The default start time is set to the closet hour.
+ * The default end time is set to the end of working hour.
+ */
+function setClosestEventTime() {
+  const workHourStartString = $('#working-hour-start').val();
+  const workHourEndString = $('#working-hour-end').val();
+
+  const userPickedDate = getUserPickedDate();
+  var defaultEventStartHour; 
+
+  if (isToday(userPickedDate)) {
+    // If user picked today to schedule for, 
+    // the default event starts on the next hour if it is before
+    // working hour ends. 
+    const now = new Date();
+    const nextHour = now.getHours() + 1;
+    defaultEventStartHour = getClosestNextHour(nextHour, workHourStartString, workHourEndString);
+  } else {
+    defaultEventStartHour = workHourStartString; 
+  }
+   
+  $('#new-event-start-time').val(defaultEventStartHour);
+  $('#new-event-end-time').val(workHourEndString);
+}
+
+/** Checks if a date is today. */
+function isToday(date) {
+  const today = new Date();
+  return (date.getFullYear() === today.getFullYear()
+      && date.getMonth() === today.getMonth()
+      && date.getDate() === today.getDate());
+}
+
+/** 
+ * Gets the closest next hour compared to the current time.
+ * This hour is displayed as the placeholder for creating new event's 
+ * start time. 
+ * If the closest hour is before working hour start time, sets it to
+ * working hour start time. If the closest hour is after working hour end
+ * time, sets it to working hour end time. 
+ * This function is called if the user picks today to schedule for. 
+ * @return closest next hour in HH:00 format. 
+ */
+function getClosestNextHour(nextHour, workHourStartString, workHourEndString) {
+  const workHourStart = parseInt(workHourStartString.split(':')[0]);
+  const workHourEnd = parseInt(workHourEndString.split(':')[0]);
+
+  var closestHour = nextHour; 
+  if (nextHour <= workHourStart) {
+    closestHour = workHourStartString;
+  } else if (nextHour >= workHourEnd) {
+    closestHour = workHourEndString;
+  } else {
+    if (nextHour < 10) {
+      closestHour = '0' + nextHour;
+    }
+    closestHour += ':00';
+  }
+
+  return closestHour; 
+}
+ 
+/** Checks the validity of the user's working hours input. */
+function checkWorkHourRange() {
+  const workHourStartHour =
+      parseInt($('#working-hour-start').val().split(':')[0]);
+  const workHourStartMinute =
+      parseInt($('#working-hour-start').val().split(':')[1]);
+  const workHourEndHour = parseInt($('#working-hour-end').val().split(':')[0]);
```

Comment:
Similarly for `$('#working-hour-end').val().split(':')`.

Reply:
Updated.